File: js/spellingItOut.js
// Create a function which takes in a word and spells it out, by consecutively adding letters
// until the full word is completed.
// https://edabit.com/challenge/Akgh37c6dccEsNBt6

const spelling = (str) => {
  const letters = [];

  for (let i = 0; i < str.length; i++) {
    letters.push(str.substring(0, i + 1));
  }

  return letters;
};

module.exports = spelling;

File: python/spelling_it_out.py
"""
Spelling it Out

Create a function which takes in a word and spells it out, by consecutively adding letters until the full word is completed.

https://edabit.com/challenge/D69v5RSgzFPqsyfwf
"""
from typing import List


def spelling(txt: str) -> List[str]:
    return [txt[:i] for i in range(1, len(txt) + 1)]
